### DISCUSSION TITLE
Class Component에 비해 Hook + Functional Component가 가지는 이점

### DISCUSSION THREAD

**Discussion by airman5573** [woowacourse-study/2022-woowahan-bansanghwe#21] [React]
[Render Props - React Docs](https://reactjs.org/docs/render-props.html)

1. 코드의 길이가 현저하게 줄어듭니다
2. 로직을 분리하기 편해집니다. HOC로 감쌀 필요 없습니다
3. this바인딩을 신경 쓸 필요가 없습니다

다만, 함수형 컴포넌트는 hook의 호출 순서를 지켜줘야 하고, life cycle이 직관적이지 않다는 단점이 있다고 생각합니다.

**Comment by InKyoJeong** [comment]
지금은 리액트를 시작하면 hook부터 당연하게쓰는데, 클래스만쓰다가 [댄 쌤이 hook을 소개하는 발표](https://youtu.be/dpw9EHDh2bM?t=694) 이거를 저자리에서 처음 들었던 사람들은 얼마나 신기했을까 생각이 들더라구요.
